### ISSUE TITLE
[Feedback](LAN) Multiplayer game host needs to set d2roffline.exe run as admin for a successful connection

### ISSUE THREAD

**Issue by 411752230** [ferib/D2R-Offline#56]
2.0.4
the original patchs.txt with only one line of Simplified Chinese language added.
0xD4AD68:9090
0xD4E25F:909090909090
0xCAFB9D:90B001
0x597E1C:90909090909090
0xC5E81C:9090C346455242: ~ MP stack corruption bypass
0xD615F2:909090909090909090909090909090909090909090909090909090: ~ show all calsses on load (shalzuth)
0x39FC03:9090909090909090909090909090909090909090: ~ allow chars to load (shalzuth)
0x1446C8:+0x270DC: ~ Chinese (simplified) *[zhCN]*

I played D2R with my friend online on a client like Gameranger. We got the same version of D2Roffline and patchs.txt. After entering the game by the client, our local ip become 0.0.0.0
![QQ截图20210419002302](https://user-images.githubusercontent.com/14128007/115152907-ce880200-a0a5-11eb-8ecc-d64e9407015c.png)

I was the host who has to set d2roffline.exe run as admin. Otherwise, my friend will crash when connecting to my game.

By the way, the Shared Stash seems never to work properly by TCP/IP. Restarting the game will lose everything left in it.(disappear)

**Comment by 411752230** [comment]
@ferib I cannot join ferib.dev. This time i am not the host but you are. My game will crash when connecting and loading. Pretty similar to what I met with my friend.

**Comment by 411752230** [comment]
How strange... What is the problem? I am able to join ferib.dev by D2RModding.exe.

**Comment by 411752230** [comment]
New finding. Able to join by 2.0.2, but 2.0.4 and 2.0.5 no.

**Comment by 411752230** [comment]
2.0.5
running log

[01:22:21.11916]: Launching game...
[01:22:21.12117]: Extra parameters not found. Proceeding...
[01:22:21.40021]: Process started...
[01:22:21.40619]: Opening process...
[01:22:21.40818]: Waiting for data at 0x7FF7818F8858...
[01:22:23.29956]: Suspending process...
[01:22:23.30154]: Process suspended
[01:22:23.30254]: Remapping process..
[01:22:23.37035]: Patching 0x7FF78036AD68
[01:22:23.37235]: Patching 0x7FF78036E25F
[01:22:23.37335]: Patching 0x7FF7802CFB9D
[01:22:23.37536]: Patching 0x7FF77FBB7E1C
[01:22:23.37536]: Patching 0x7FF78027E81C
[01:22:23.37636]: Patching 0x7FF7803815F2
[01:22:23.37736]: Patching 0x7FF77F9BFC03
[01:22:23.37835]: Patching 0x7FF77F7646C8
[01:22:23.38437]: Bypassed CRC at 7FF77F6D54D0
[01:22:23.38533]: Bypassed CRC at 7FF77F6D7430
[01:22:23.38735]: Bypassed CRC at 7FF77F712F50
[01:22:23.38833]: Bypassed CRC at 7FF77F71EE80
[01:22:23.38932]: Bypassed CRC at 7FF77F71F550
[01:22:23.39234]: Bypassed CRC at 7FF77F7862B1
[01:22:23.39431]: Bypassed CRC at 7FF77F7DD901
[01:22:23.42925]: Bypassed CRC at 7FF7802358C0
[01:22:23.43222]: Bypassed CRC at 7FF780235DB0
[01:22:23.43322]: Bypassed CRC at 7FF780242251
[01:22:23.43520]: Bypassed CRC at 7FF7802428D0
[01:22:23.45415]: Bypassed CRC at 7FF780808111
[01:22:23.46513]: Bypassed CRC at 7FF780B3C501
[01:22:23.46613]: Bypassed CRC at 7FF780B5CFD1
[01:22:23.46815]: Bypassed CRC at 7FF780B6FC71
[01:22:23.47012]: Bypassed CRC at 7FF780B87891
[01:22:23.47112]: Bypassed CRC at 7FF780B8F690
[01:22:23.47214]: Bypassed CRC at 7FF780B95FC0
[01:22:23.47311]: Bypassed CRC at 7FF780BB23A0
[01:22:23.47512]: Bypassed CRC at 7FF780BBD9B1
[01:22:23.47713]: Bypassed CRC at 7FF780BC7DD1
[01:22:23.47816]: Bypassed CRC at 7FF780BE4791
[01:22:23.47916]: Bypassed CRC at 7FF780C0EA30
[01:22:23.48013]: Bypassed CRC at 7FF780C12D31
[01:22:23.48113]: Bypassed CRC at 7FF780C1FB61
[01:22:23.48212]: Bypassed CRC at 7FF780C33930
[01:22:23.48312]: Bypassed CRC at 7FF780C3C8F1
[01:22:23.48412]: Bypassed CRC at 7FF780C3E981
[01:22:23.48514]: Bypassed CRC at 7FF780C502E0
[01:22:23.48611]: Bypassed CRC at 7FF780C542F0
[01:22:23.48611]: Bypassed CRC at 7FF780C5CAE1
[01:22:23.48711]: Bypassed CRC at 7FF780C60771
[01:22:23.48811]: Bypassed CRC at 7FF780C63F81
[01:22:23.48910]: Bypassed CRC at 7FF780C65D71
[01:22:23.49013]: Bypassed CRC at 7FF780C7AFF1
[01:22:23.49013]: Bypassed CRC at 7FF780C80A50
[01:22:23.49309]: Bypassed CRC at 7FF780CA83B1
[01:22:23.49414]: Bypassed CRC at 7FF780CB8961
[01:22:23.52502]: Resuming process..
[01:22:23.52800]: Done!
[01:22:23.52900]: Press any key to exit...

**Comment by madchenxing** [comment]
If you're the host, use the ip that gameranger gives you.  Make sure your friend(clinet) is in the same room same line.

**Comment by 411752230** [comment]
> If you're the host, use the ip that gameranger gives you. Make sure your friend(clinet) is in the same room same line.

Thanks for the reply. I knew what you suggest. But the problem is as a host you have to run d2roffline as admin.

**Comment by madchenxing** [comment]
I think its a gameranger problem...Cause it doesnt happen without gameranger.

**Comment by 411752230** [comment]
> I think its a gameranger problem...Cause it doesnt happen without gameranger.

No. It should be an issue. 2.0.2 work flawlessly with gameranger and in joining ferib.dev. But 2.0.4 and 2.0.5 do have the crash bug.

**Comment by ferib** [comment]
v2.0.7 should fix this.

**Comment by 411752230** [comment]
> v2.0.7 should fix this.

Cool! Will check.
